### PR TITLE
Rename CMake project to gz

### PR DESCRIPTION
**Pull request by chapulina**
* See https://github.com/ignition-tooling/release-tools/issues/718

### CODE REVIEW

**Comment by scpeters** [comment]
may need to replace `ignition-` with `gz-` in the following script to fix some tests:

* https://github.com/ignitionrobotics/ign-cmake/blob/gz_cmake_project/examples/test_core_child_requires_core_no_deps.bash

**Comment by scpeters** [comment]
> may need to replace `ignition-` with `gz-` in the following script to fix some tests:
> 
> * https://github.com/ignitionrobotics/ign-cmake/blob/gz_cmake_project/examples/test_core_child_requires_core_no_deps.bash

done in https://github.com/ignitionrobotics/ign-cmake/pull/244/commits/e54db0df90d24453a9a202798e7931be70ec45c0 and added an extra fix to the pkg-config file name in https://github.com/ignitionrobotics/ign-cmake/pull/244/commits/12be647e6e428af7af5eeee4573f21c7a40e01f5

**Comment by scpeters** [comment]
I just noticed there is an `ignition-` that should change to `gz-` on the following line, but there's also a super weird `1.1` that makes no sense to me

* https://github.com/ignitionrobotics/ign-cmake/blob/gz_cmake_project/cmake/pkgconfig/ignition-component.pc.in#L7

the `1.1` is also in [cmake/pkgconfig/ignition.pc.in](https://github.com/ignitionrobotics/ign-cmake/blob/gz_cmake_project/cmake/pkgconfig/ignition.pc.in#L8)

File: cmake/ignition-config.cmake.in [hunk 1]
```diff
@@ -58,9 +58,9 @@ set(@PKG_NAME@_CONFIG_INCLUDED TRUE)
 
 # Get access to the find_dependency utility
 include(CMakeFindDependencyMacro)
-# Find ignition-cmake, because we need its modules in order to find the rest of
+# Find gz-cmake, because we need its modules in order to find the rest of
 # our dependencies.
-find_dependency(ignition-cmake@IGNITION_CMAKE_VERSION_MAJOR@)
+find_dependency(@PROJECT_NAME@)
```
Comment (scpeters) [inline cmake/ignition-config.cmake.in:63]: this is a mistake

Comment (scpeters) [reply cmake/ignition-config.cmake.in]: fixed in https://github.com/ignitionrobotics/ign-cmake/pull/244/commits/e44d9dc0bfd2c715238247d6465a7447cb8b0089

Comment (scpeters) [reply cmake/ignition-config.cmake.in]: fixed in https://github.com/ignitionrobotics/ign-cmake/pull/244/commits/e44d9dc0bfd2c715238247d6465a7447cb8b0089

**Comment by scpeters** [comment]
> I just noticed there is an `ignition-` that should change to `gz-` on the following line, but there's also a super weird `1.1` that makes no sense to me
> 
> * https://github.com/ignitionrobotics/ign-cmake/blob/gz_cmake_project/cmake/pkgconfig/ignition-component.pc.in#L7
> 
> the `1.1` is also in [cmake/pkgconfig/ignition.pc.in](https://github.com/ignitionrobotics/ign-cmake/blob/gz_cmake_project/cmake/pkgconfig/ignition.pc.in#L8)

some explanation for `1.1` in https://osrf-migration.github.io/ignition-gh-pages/#!/ignitionrobotics/ign-cmake/pull-requests/104/page/1

I think we can remove it

**Comment by chapulina** [comment]
> regex replaces removing ignition to no-op

Good point, removed in [a477d3b](https://github.com/gazebosim/gz-cmake/pull/244/commits/a477d3b38a0b059fa3ba8e79ca914d4adb7a249d) since this PR will go along all downstream project name changes anyway.

**Comment by chapulina** [comment]
> Good point, removed in [a477d3b](https://github.com/gazebosim/gz-cmake/pull/244/commits/a477d3b38a0b059fa3ba8e79ca914d4adb7a249d) since this PR will go along all downstream project name changes anyway.

Ah that broke some test cases. Reverted in [73ef90e](https://github.com/gazebosim/gz-cmake/pull/244/commits/73ef90eff1c88e0cb55dc13ba82484e2acc056fb). We can address that in follow-up PRs.

**Review by methylDragon — APPROVED** [review]

**Pull request merged**